src/renderer/utils.js

```
@@ -339,10 +352,159 @@ const video = {
         }
     }
 }
+const acctOps = {
+    async login(data) {
+        switch(data.accountType) {
+            case "hive" : {
+                try { 
+                    const userAccounts = await hive.api.getAccountsAsync([data.username]);
+                    console.log(userAccounts)
+                    const pubWif = userAccounts[0].posting.key_auths[0][0];
+                    const wif = hive.auth.toWif(data.key)
+    
+                    const Valid = hive.auth.wifIsValid(data.key, pubWif);
+    
+                    if(Valid){
+    
+                        const profile = {
+                            _id: userAccounts[0].id.toString(),
+                            nickname: data.profile,
+                            keyring: [
+                                {
+                                    type: 'hive',
+                                    username: data.username,
+                                    public: {
+                                        pubWif
+                                    },
+                                    encrypted: data.encrypted,
+                                    private: {
+                                        key: encryptWithAES(data.key)
+                                    }
+                                }
+                            ]
+                        }
+                        const profileID = profile._id;
+                        const check_profile = (await PromiseIPC.send("accounts.has", profileID));
+                        if (check_profile) {
+                            return {
+                                message: "Account exists already"
+                            }
```

Throw an error instead of returning an object